sockeye/vocab.py

```diff
@@ -320,65 +332,68 @@ def load_or_create_vocabs(source_paths: List[str],
+                                                            min_count=word_min_count_source,
+                                                            pad_to_multiple_of=pad_to_multiple_of,
+                                                            mapper=mapper)
 
         else:
             vocab_path = source_vocab_path if source_vocab_path is not None else target_vocab_path
             logger.info("Using %s as a shared source/target vocabulary." % vocab_path)
             vocab_source = vocab_target = vocab_from_json(vocab_path)
 
     else:
-        vocab_source = load_or_create_vocab(source_path, source_vocab_path, num_words_source, word_min_count_source,
-                                            pad_to_multiple_of=pad_to_multiple_of)
-        vocab_target = load_or_create_vocab(target_path, target_vocab_path, num_words_target, word_min_count_target,
-                                            pad_to_multiple_of=pad_to_multiple_of)
+        vocab_source = load_or_create_vocab(shard_source_path, source_vocab_path, num_words_source, word_min_count_source,
+                                            pad_to_multiple_of=pad_to_multiple_of, mapper=mapper)
+        vocab_target = load_or_create_vocab(shard_target_path, target_vocab_path, num_words_target, word_min_count_target,
+                                            pad_to_multiple_of=pad_to_multiple_of, mapper=mapper)
 
     vocab_source_factors = []  # type: List[Vocab]
-    if source_factor_paths:
+    if shard_source_factor_paths:
         logger.info("(2) Additional source factor vocabularies")
         if len(source_factor_vocab_same_as_source) > 1:
-            utils.check_condition(len(source_factor_vocab_same_as_source) == len(source_factor_paths),
+            utils.check_condition(len(source_factor_vocab_same_as_source) == len(shard_source_factor_paths),
                                   "The number of flags for sharing the vocabulary of "
                                   "source factors does not match the number of source "
                                   "factors.")
         elif len(source_factor_vocab_same_as_source) == 1:
-            source_factor_vocab_same_as_source = source_factor_vocab_same_as_source * len(source_factor_paths)
+            source_factor_vocab_same_as_source = source_factor_vocab_same_as_source * len(shard_source_factor_paths)
         else:
-            source_factor_vocab_same_as_source = [False] * len(source_factor_paths)
+            source_factor_vocab_same_as_source = [False] * len(shard_source_factor_paths)
 
-    for factor_path, factor_vocab_path, share_source_vocab in zip(source_factor_paths,
-                                                                  source_factor_vocab_paths,
-                                                                  source_factor_vocab_same_as_source):
+    for shard_factor_path, factor_vocab_path, share_source_vocab in zip(shard_source_factor_paths,
```

Comment:
shard_source_factor_paths is a list of shard_source_factor_paths[factor][shard], right?
Could you add this as a comment and `make shard_factor_path` plural, so `shard_factor_paths` just to make it 'clear' that it is a list.